crawl-ref/source/mon-info.cc

```diff
@@ -1825,6 +1783,533 @@ void get_monster_info(vector<monster_info>& mons)
     sort(mons.begin(), mons.end(), monster_info::less_than_wrapper);
 }
 
+void mons_to_string_pane(string& desc, int& desc_colour, bool fullname,
+                         const vector<monster_info>& mi, int start,
+                         int count)
+{
+    mi[start].to_string(count, desc, desc_colour, fullname, nullptr, false);
+    mons_conditions_string(desc, mi, start, count, true);
+}
+
+void mons_conditions_string(string& desc, const vector<monster_info>& mi,
+                            int start, int count, bool listall)
+{
+    static vector<monster_info_func> monster_info_funcs =
+    init_monster_info_funcs();
+
+    ostringstream out;
+    if (listall)
+        out << desc;
+    bool first=true; bool didany = false;
+    for (auto &info: monster_info_funcs) {
+        int num = 0;
+        for (int j = start; j < start+count; j++)
+        {
+            if (info.test(mi[j], listall))
+            {
+                if (!listall)
+                {
+                    out << info.singular;
+                    desc=out.str();
+                    return;
+                }
+                else
+                    num++;
+            }
+        }
+        if (num > 0)
+        {
+            if (first)
+            {
+                first = false;
+                out << " (";
+                didany = true;
+            }
+            else
+                out << ", ";
+            if (num < count)
+                out << num << " ";
+            out << (num > 1 ? info.plural : info.singular);
+        }
+    }
+    if (didany)
+        out << ")";
+    desc = out.str();
+}
+
+vector<monster_info_func> init_monster_info_funcs()
+{
+    vector<monster_info_func> toret; // well, I couldn't think of a good name
+    toret.push_back({"charmed", "charmed",
+            [](const monster_info &mi, bool newconditions)
+            {
+                return newconditions && mi.is(MB_CHARMED);
+            }
+        });
+// I can't think of a good short word here
+    toret.push_back({"hexed", "hexed",
+            [](const monster_info &mi, bool newconditions)
+            {
+                return newconditions && mi.is(MB_HEXED);
+            }
+        });
+    toret.push_back({"berserk", "berserk",
+            [](const monster_info &mi, bool newconditions)
+            {
+                return mi.is(MB_BERSERK)
+                && (!mi.is(MB_SLOWED) || !newconditions);
+            }
+        });
+    toret.push_back({"berserk+slow", "berserk+slow",
+            [](const monster_info &mi, bool newconditions)
+            {
+                return mi.is(MB_BERSERK)
+                && mi.is(MB_SLOWED) && newconditions;
+            }
+        });
+    toret.push_back({"fast", "fast",
+            [](const monster_info &mi, bool newconditions)
+            {
+                return newconditions && mi.is(MB_HASTED)
+                && !mi.is(MB_BERSERK) && !mi.is(MB_SLOWED);
+            }
+        });
```

Comment:
Strange that `fast+slow` is missing but `berserk+slow` is present.

Reply:
I left out fast+slow because it's normal right now. Admittedly, it could go in as a warning the monster's speed is likely to change in the near future, but a concern I have been wildly inconsistent about is cluttering it up with less important information.

Berserk+slow tells me the monster still hits hard but isn't fast like a normal berserk monster. Arguably it could be rolled into "strong" by virtue of something like the rationale above about showing the status right now.